tests/unit/metrics/test_log.py

```diff
@@ -0,0 +1,172 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2018 Spotify AB
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numbers
+
+import pytest
+
+from gordon.metrics import log
+
+
+@pytest.mark.parametrize('context,context_str', [
+    ({'source': 'external'}, ' context: {\'source\': \'external\'}'),
+    ({}, '')
+])
+def test_logger_adapter_log_outputs_message(caplog, context, context_str):
+    """Metric is formatted and output to log."""
+    logger_adapter = log.LoggerAdapter('info')
+    metric = {
+        'metric_name': 'events',
+        'value': 12.12,
+        'context': context
+        }
+    logger_adapter.log(metric)
+    assert 1 == len(caplog.records)
+    expected_log = '[events] value: 12.12' + context_str
+    assert expected_log == caplog.records[0].msg
+
+
+@pytest.fixture
+def mock_logger_adapter(mocker):
+    return mocker.Mock(log.LoggerAdapter)
+
+
+@pytest.fixture
+def log_timer(mock_logger_adapter):
+    metric = {'metric_name': 'test-metric', 'value': 13.05, 'context': {}}
+    return log.LogTimer(metric, mock_logger_adapter, 's')
+
+
+@pytest.mark.asyncio
+async def test_log_timer_manual(log_timer):
+    """Times when using start/stop interface."""
+
+    await log_timer.start()
+    assert isinstance(log_timer._start_time, numbers.Number)
+
+    await log_timer.stop()
+    assert 1 == log_timer.logger.log.call_count
+    actual_metric = log_timer.logger.log.mock_calls[0][1][0]
+    assert 'test-metric' == actual_metric['metric_name']
+    assert {} == actual_metric['context']
+    assert 0 < actual_metric['value']
+
+
+@pytest.mark.asyncio
+async def test_log_timer_as_context_manager(log_timer):
+    """Times when using as a context manager."""
+    async with log_timer as t:
+        assert isinstance(t._start_time, numbers.Number)
+    actual_metric = log_timer.logger.log.mock_calls[0][1][0]
+    assert 1 == log_timer.logger.log.call_count
+    assert 'test-metric' == actual_metric['metric_name']
+    assert {} == actual_metric['context']
+    assert 0 < actual_metric['value']
+
+
+@pytest.fixture
+def relay_config():
+    return {
+        'time_unit': 's',
+        'log_level': 'info',
+    }
```

Comment:
What do you think about adding tests for config-related, i.e. `inof` -> Attribute error (maybe something more graceful should be thrown?), `INFO` -> does not throw an attribute error, `time_unit: fff` -> key error, etc.

Reply:
I'd prefer to leave this work to the config validator that's in the works.